Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -341,6 +348,40 @@ def audit_formula_name
       openssl@1.1
     ].freeze
 
+    def audit_license
+      if !formula.license.blank?
+        if @spdx_ids.key?(formula.license)
+          return unless @online
+
+          user, repo = get_repo_data(%r{https?://github\.com/([^/]+)/([^/]+)/?.*}, false)
+          return if user.nil?
+
+          github_license = get_repo_license_data(user, repo)
+          return if github_license && (github_license == formula.license)
+
+          problem "License mismatch - Github license is: #{github_license}, "\
+          "but Formulae license states: #{formula.license}."
+        else
+          problem "#{formula.license} is not a standard SPDX license id."
+        end
+      else
+        problem "No license specified for package."
```

Comment:
We probably want this to be a `@new_formula` audit for now, at least until we get all licensing information added to Homebrew/homebrew-core. As in, it should check that the stated license matches the github license, and that the stated license is on our license allowlist, but it shouldn't squawk for missing licenses unless we're running `brew audit --new-formula`.